### ISSUE TITLE
Crystal compiler bug (UInt128 as Hex)

### ISSUE THREAD

**Issue by delef** [crystal-lang/crystal#7449]
`0xadc0db298a7401251f4fa96ba429cb24`
https://carc.in/#/r/6a3j

```
Overflow (OverflowError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from __crystal_main
  from main
  from __libc_start_main
  from _start
  from ???
Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues
```

in addition to: #7448 

I create this issue, because another exception is raised.

**Comment by jhass** [comment]
With stacktrace:

```cr
0xadc0db298a7401251f4fa96ba429cb24
```

```
Arithmetic overflow (OverflowError)
  from projects/crystal/src/compiler/crystal/syntax/lexer.cr:1728:23 in 'scan_hex_number'
  from projects/crystal/src/compiler/crystal/syntax/lexer.cr:1625:9 in 'scan_zero_number'
  from projects/crystal/src/compiler/crystal/syntax/lexer.cr:1622:5 in 'scan_zero_number'
  from projects/crystal/src/compiler/crystal/syntax/lexer.cr:756:9 in 'next_token'
  from projects/crystal/src/compiler/crystal/syntax/parser.cr:5830:7 in 'next_token'
  from projects/crystal/src/compiler/crystal/syntax/lexer.cr:2881:7 in 'next_token_skip_statement_end'
  from projects/crystal/src/compiler/crystal/syntax/parser.cr:75:21 in 'parse'
  from projects/crystal/src/compiler/crystal/compiler.cr:242:7 in 'parse'
  from projects/crystal/src/compiler/crystal/compiler.cr:130:9 in 'parse'
  from projects/crystal/src/compiler/crystal/compiler.cr:165:7 in 'compile'
  from projects/crystal/src/compiler/crystal/command.cr:266:7 in 'compile'
  from projects/crystal/src/compiler/crystal/command.cr:264:5 in 'compile'
  from projects/crystal/src/compiler/crystal/command.cr:161:5 in 'build'
  from projects/crystal/src/compiler/crystal/command.cr:67:16 in 'run'
  from projects/crystal/src/compiler/crystal/command.cr:46:5 in 'run'
  from projects/crystal/src/compiler/crystal/command.cr:45:3 in 'run'
  from projects/crystal/src/compiler/crystal.cr:8:1 in '__crystal_main'
  from projects/crystal/src/crystal/main.cr:97:5 in 'main_user_code'
  from projects/crystal/src/crystal/main.cr:86:7 in 'main'
  from projects/crystal/src/crystal/main.cr:106:3 in 'main'
  from __libc_start_main
  from _start
  from ???
```

**Comment by straight-shoota** [comment]
This has been resolved by #11571
